drgn_tools/task.py

```diff
@@ -161,3 +170,91 @@ def task_cpu(task: drgn.Object) -> int:
     if has_member(task, "cpu"):
         return task.cpu.value_()
     return task_thread_info(task).cpu.value_()
+
+
+def format_nanosecond_duration(nanosecs: int) -> str:
+    """
+    :returns: conversion of nanoseconds to [dd hh:mm:ss.ms] format
+    """
+    secs = nanosecs_to_secs(nanosecs)
+    dd, rem = divmod(secs, 86400)
+    hh, rem = divmod(rem, 3600)
+    mm, secs = divmod(rem, 60)
+    return "%02ld %02ld:%02ld:%.3f" % (dd, hh, mm, secs)
+
+
+def get_pid(task: Object) -> int:
+    """
+    :returns: PID of the task
+    """
+    return task.pid.value_()
+
+
+def get_command(task: Object) -> str:
+    """
+    :returns: name of the command
+    """
+    return escape_ascii_string(task.comm.string_())
+
+
+def show_tasks_last_runtime(prog: Program, tasks: Iterable[Object]) -> None:
+    """
+    Display task information in their last arrival order.
+    """
+    print(
+        "LAST_ARRIVAL".ljust(25)
+        + "STATE".ljust(18)
+        + "PID".ljust(20)
+        + "TASK".ljust(20)
+        + "CPU".ljust(15)
+        + "COMMAND".ljust(20)
+    )
+    tasklist = []
+    for t in tasks:
+        cpu = task_cpu(t)
+        pid = get_pid(t)
+        state = task_state_to_char(t)
+        task = find_task(prog, pid)
```

Comment:
This is unnecessary, `t` is already the task struct, you can simply use `hex(t.value_())` below.